python/lsst/dax/imgserv/imageREST_v1.py

```diff
@@ -0,0 +1,226 @@
+
+import lsst.log as log
+
+from http.client import BAD_REQUEST, INTERNAL_SERVER_ERROR, NOT_FOUND
+from .locateImage import (
+        image_open_v1, W13DeepCoaddDb, W13RawDb, W13CalexpDb
+)
+
+from .image_v1 import Image
+from .dispatch_v1 import Dispatcher
+from .hashutil import Hasher
+
+imageRESTv1 = Blueprint("imageRESTv1", __name__, static_folder="static",
+                      template_folder="templates")
+
+# To be called from webserv
+def load_imgserv_config(config_path, db_auth_conf):
+    """Load configuration info into ImageServ."""
+    if config_path is None:
+        # use default root_path for imageRESTv1
+        config_path = imageRESTv1.root_path+"/config/"
+    f_json = os.path.join(config_path, "settings.json")
+    # load the general config file  
+    current_app.config.from_json(f_json)
+    # configure the log file (log4cxx)
+    log.configure(os.path.join(config_path, "log.properties"))
+    current_app.config["DAX_IMG_DBCONF"] = db_auth_conf
+    current_app.config["DAX_IMG_CONFIG"] = config_path
+
+@imageRESTv1.route("/")
+def index():
+    return make_response(render_template("index_v1.html"))
+
+@imageRESTv1.route("/availability", methods=["GET"])
+def getimage_avail():
+    return _getimage_avail(request)
+
+@imageRESTv1.route("/capabilities", methods=["GET"])
+def getimage_capabilities():
+    return _getimage_capabilities(request)
+
+@imageRESTv1.route("/<db_id>", methods=["GET"])
+def getimage_sync(db_id):
+    return _getimage(request, db_id)
+
+@imageRESTv1.route("/<db_id>/async", methods=["POST"])
+def getimage_async(db_id):
+    return _getimage_async(request)
+
+def _getimage_avail(_req):
+    """Return availability status."""
+    fmt  = _req.accept_mimetypes.best_match(["application/json",
+        "text/html"])
+    if fmt == "text/html":
+        return "DAX Image Web Service <p> v0: service is accepting queries. <p>\
+                v1: service is accepting queries."
+    else:
+        json.dumps({
+            "availability": {
+                "api": [ "v0", "v1" ],
+                "v0": {
+                    "status": "dax_imgserv_v0 is accepting queries",
+                    'available': 'true'
+                },
+                "v1": {
+                    "status": "dax_imgserv_v1 is accepting queries",
+                    "available": "true"
+                }
+            }
+        })
+
+
+def _getimage_capabilities(_req):
+    """Return capabilities of this service."""
+    fmt  = _req.accept_mimetypes.best_match(["application/json",
+        "text/html"])
+    if fmt == "text/html":
+        return "Image Web Service <p> <a href='image/v0/DC_W13_Stripe82'> API v0 - DB: W13_sdss \
+                Interface: DAX <p> <a href='image/v1/DC_W13_Stripe82'>API v1 - \
+                DB: W13_sdss &nbsp; Interface: DAX </a>"
+    else:
+        json.dumps({
+            "capabilities": {
+                "api": [ "v0", "v1" ],
+                "v0": {
+                    "db": "W13_sdss",
```

Comment:
Not sure about this, we may need to discuss. I don't think this is the intended use for capabilities, and it'd be better to not implement it